code/game/machinery/spaceheater.dm

```diff
@@ -80,39 +81,48 @@
 				update_icon()
 			return
 
-		var/datum/gas_mixture/env = L.return_air()
+		for(var/turf/T in block(
+				locate(
+					max(L.x - heating_range, 1), max(L.y - heating_range, 1), L.z
+				), locate(
+					min(L.x + heating_range, world.maxx), min(L.y + heating_range, world.maxy), L.z
+				)))
+			heat_turf(T)
+	else
+		on = FALSE
+		update_icon()
+		return PROCESS_KILL
```

Comment:
Instead of doing this make heater need to be secured to the floor.
After being secured it should build list of (valid!) turfs around it and iterate over the list instead.